src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -68,11 +68,6 @@ public override bool Equals(object comparand)
                 return false;
             }
 
-            if (_address.GetHashCode() != address._address.GetHashCode())
-            {
-                return false;
```

Comment:
Ah, I hadn't noticed the hash caching going on.

Does this micro-optimization really make sense though, considering that Equals just needs to compare two 6-byte arrays? I mean, would it make sense to do something similar when comparing two 8-byte longs?